particles/Tutorial/Kotlin/4_Handles/GetPerson.kt

```diff
@@ -0,0 +1,30 @@
+package arcs.tutorials
+
+import arcs.*
+import kotlin.native.internal.ExportForCppRuntime
+
+
+/**
+ * Sample WASM Particle.
+ */
+class GetPersonParticle : Particle() {
+    private val person = Singleton { GetPerson_Person() }
+
+    override fun getTemplate(slotName: String) = """
+        <input placeholder="Enter your name" spellcheck="false" on-change="onNameInputChange">
+        <div slotid="greetingSlot"></div>""".trimIndent()
+
+    init {
+      registerHandle("person", person)
+
+      eventHandler("onNameInputChange") {
+        val p = person.get() ?: GetPerson_Person()
+        p.name = it["value"] ?: "Human"
+        person.set(p)
+      }
+    }
+}
+
+@Retain
+@ExportForCppRuntime()
+fun _newGetPerson() = GetPersonParticle().toWasmAddress()
```

Comment:
Side note: I'm starting to think we should consider writing an annotation processor for particles that automagically generates a file with a bunch of these `@Retain` things before compilation.  It'd be really nice to not require this boilerplate from clients.

Reply:
CC: @piotrswigon

Reply:
I would agree. Added to #3665

Reply:
Definitely. @SHeimlich feel free to give it a shot at some point this month. One comment would be to try to do it in a way that's easy to execute both in the internal and external repos.